src/site/content/en/blog/google-search-sw/index.md

```diff
@@ -0,0 +1,463 @@
+
+Additionally, the team found that, based on measuring service worker boot times
+on real-world devices, there was a wide distribution of startup times, with some
+low-end mobile devices taking almost as much time to start up the service worker
+as it might take to make the network request for the results page's HTML.
+
+**Solution**
+
+The single, most crucial feature that allowed the Google Search team to move
+ahead with their service worker launch is
+[navigation preload](https://developers.google.com/web/updates/2017/02/navigation-preload).
+Using navigation preload is a key performance win for any service worker that
+needs to use a response from the network to satisfy navigation requests. It
+provides a hint to the browser to start making the navigation request
+immediately, at the same time as the service worker starts up. As long as the
+amount of time it takes for the service worker to start up is less than the
+amount of time it takes to get a response back from the network, it means that
+there should not be any latency overhead introduced by the service worker.
+
+![An illustration of the SW startup done in parallel with the navigation request.](with-preload.png)
+
+The Search team also needed to avoid using a service worker on low-end mobile
+devices where that service worker boot time could exceed the time it would take
+to get back a network response. Since there's no hard-and-fast rule for what
+constitutes a "low-end" device, they came up with the heuristic of
+[checking the total RAM](https://developers.google.com/web/updates/2017/12/device-memory)
+installed on the device. Anything less than 2 gigabytes of memory fell into
+their low-end device category, where based on the real-world experiments they've
+run, service worker startup time would be unacceptable.
+
+Available storage space is another consideration, since the full set of
+resources to be cached for future use can run to several megabytes. The
+[`navigator.storage` interface](https://developers.google.com/web/updates/2017/08/estimating-available-storage-space)
+comes in handy there, allowing the Google Search page to figure out in advance
+whether their attempts to cache data run the risk of failing due to storage
+quota failures.
+
+This left the Search team with multiple pieces of criteria that they could use
+to determine whether or not to use a service worker: if a user comes to the
+Google Search page using a browser that supports navigation preload, and has at
+least 2 gigabytes of RAM, and enough free storage space, then a
+[service worker will be registered](https://developers.google.com/web/fundamentals/primers/service-workers/lifecycle#the_first_service_worker).
+Browsers or devices that don't meet that criteria won't end up with a service
+worker, but they'll still see the same Google Search experience as they always
+have.
+
+One side benefit of this selective registration is that it has allowed the
```

Comment:
Slight rewrites for two sentences:

One side benefit of this selective registration is the ability to ship a smaller, more efficient service worker. Targeting fairly modern browsers to run the service worker code eliminates the overhead of transpilation and polyfills for older browsers.